adov_flutter/lib/screens/home_screen/home_screen.dart

```diff
@@ -2,16 +2,38 @@ import 'package:adov_flutter/app.dart';
 import 'package:adov_flutter/screens/home_screen/conference_sponsors.dart';
 import 'package:adov_flutter/screens/home_screen/conference_stats.dart';
 import 'package:adov_flutter/screens/home_screen/next_talk.dart';
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 
 import '../../app.dart';
 import '../../style.dart';
 import 'conference_details.dart';
 
-class HomeScreen extends StatelessWidget {
+class HomeScreen extends StatefulWidget {
+
+  @override
+  State createState() => HomeScreenState();
+}
+
+class HomeScreenState extends State<HomeScreen> {
+  bool isLoggedIn;
+
+
+  @override
+  void initState() {
+    isLoggedIn = false;
+    FirebaseAuth.instance.authStateChanges().listen((user) {
+      setState(() {
+        isLoggedIn = user != null;
+      });
+    });
+    super.initState();
```

Comment:
explain this please

Reply:
isso regista um `listener` para o estado de autenticação do Firebase Auth, sempre que houver uma alteração (login ou logout no nosso caso) executa o setState() atualizado a flag `isLoggedIn` de acordo. Isto permite ter alterações dinamicas no ecrâ. Por exemplo se o user estiver loggado queremos que apareçam opções para editar talks e logout, mas caso não esteja logado apenas queremos que veja opção para iniciar sessão.